packages/xstate-svelte/src/useSelector.ts

```diff
@@ -0,0 +1,30 @@
+import { get, readable } from 'svelte/store';
+import type { ActorRef, Subscribable, Subscription } from 'xstate';
+
+const defaultCompare = (a, b) => a === b;
+
+export const useSelector = <
+  TActor extends ActorRef<any, any>,
+  T,
+  TEmitted = TActor extends Subscribable<infer Emitted> ? Emitted : never
+>(
+  actor: TActor,
+  selector: (emitted: TEmitted) => T,
+  compare: (a: T, b: T) => boolean = defaultCompare
```

Comment:
There is a missing 4th argument `getSnapshot` compared to `@xstate/react`.

Reply:
I think it's fine for that to be added later or emitted; actors passed into `useSelector` should adhere to the `ActorRef` interface (i.e., it's not compatible with other things), so when I added `getSnapshot` as an option, I think I was being too flexible 😓